tests/usm/usm_api.h

```diff
@@ -0,0 +1,688 @@
+
+    detail::check_values<count>(ptr, reference);
+  }
+};
+
+/** @brief Provides test logic for the memset() method tests
+ */
+template <typename T, size_t count>
+class memset {
+  const int value;
+
+ public:
+  static constexpr size_t size = count * sizeof(T);
+
+  memset(sycl::queue &, int seed) : value(seed) {}
+
+  /** @brief The memset() method doesn't support non-USM pointers
+   */
+  static constexpr bool has_non_usm_support() { return false; }
+
+  template <allocation alloc>
+  static std::string description() {
+    return "memset using " + get_allocation_decription<alloc>();
+  }
+
+  template <typename parentT, typename... depEventsT>
+  auto call(parentT &parent, T *ptr, depEventsT &&... events) const {
+    return parent.memset(ptr, value, size, std::forward<depEventsT>(events)...);
+  }
+
+  void check(const T *ptr) const {
+    T reference[count];
+    std::memset(reference, value, size);
+
+    detail::check_values<count>(ptr, reference);
+  }
+};
+
+/** @brief Provides test logic for the prefetch() method tests
+ */
+template <typename T, size_t count>
+class prefetch {
+ public:
+  static constexpr size_t size = count * sizeof(T);
+
+  prefetch(sycl::queue &, int) {}
+
+  /** @brief The prefetch() method doesn't support non-USM pointers
+   */
+  static constexpr bool has_non_usm_support() { return false; }
+
+  template <allocation alloc>
+  static std::string description() {
+    return "prefetch using " + get_allocation_decription<alloc>();
+  }
+
+  template <typename parentT, typename... depEventsT>
+  auto call(parentT &parent, T *ptr, depEventsT &&... events) const {
+    return parent.prefetch(ptr, size, std::forward<depEventsT>(events)...);
+  }
+
+  void check(const T *) const {
+    // Results are implementation-defined
+  }
+};
+
+/** @brief Provides test logic for the mem_advise() method tests
+ */
+template <typename T, size_t count>
+class mem_advise {
+ public:
+  static constexpr size_t size = count * sizeof(T);
+
+  mem_advise(sycl::queue &, int) {}
+
+  /** @brief The mem_advise() method doesn't support non-USM pointers
+   */
+  static constexpr bool has_non_usm_support() { return false; }
+
+  template <allocation alloc>
+  static std::string description() {
+    return "mem_advise using " + get_allocation_decription<alloc>();
+  }
+
+  template <typename parentT, typename... depEventsT>
+  auto call(parentT &parent, T *ptr, depEventsT &&... events) const {
+    const int advice = 0;  // Reset to defaults according to the SYCL 2020 spec
+    return parent.mem_advise(ptr, size, advice,
+                             std::forward<depEventsT>(events)...);
+  }
+
+  void check(const T *) const {
+    // Results are implementation-defined
+  }
+};
+
+}  // namespace tests
+
+/** @brief Kernel name for event_generator::init() method
+ */
+template <typename T, size_t gen_buf_size>
+struct init_kernel_name;
+
+/** @brief Kernel name for event_generator::copy_arrays() method
```

Comment:
```suggestion
/** @brief Kernel name for event_generator::copy_arrays() member function
```

Reply:
I've replaced "method" to "member function" in this commit: 800816219c8f8b2a87887df4ef54c8088b0a153b